pkg/apis/config/v1alpha1/types.go

```diff
@@ -0,0 +1,264 @@
+// Copyright The NRI Plugins Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/containers/nri-plugins/pkg/apis/config/v1alpha1/instrumentation"
+	"github.com/containers/nri-plugins/pkg/apis/config/v1alpha1/log"
+	"github.com/containers/nri-plugins/pkg/apis/config/v1alpha1/resmgr/control"
+	"github.com/containers/nri-plugins/pkg/apis/config/v1alpha1/resmgr/policy/balloons"
+	"github.com/containers/nri-plugins/pkg/apis/config/v1alpha1/resmgr/policy/template"
+	"github.com/containers/nri-plugins/pkg/apis/config/v1alpha1/resmgr/policy/topologyaware"
+)
+
+// TopologyAwareConfig represents the configuration for the topology-aware policy.
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +genclient
+type TopologyAwareConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	TopologyAware topologyaware.Config `json:"topologyAware"`
```

Comment:
Suggesting  `json:"topology-aware"` because users see the policy name written with `-` in documentation, helm commands, deployment files and binary name, too.

Reply:
(Hesitating... looking at TopologyAwareConfig CRs (like topologyaware-config.yaml), having a hyphen there in "topologyAware:" would not really fit in the picture too well, if it is even allowed. Probably better keep it without hyphen after all.)

Reply:
let's forget about this, bad idea after all, based on first feeling when trying to play with config files.